### ISSUE TITLE
Persistant Adapters at adapter registration time

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11294]
|<img alt="ironfroggy's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ironfroggy reported|
|-|-|
|Trac ID|trac#556|
|Type|enhancement|
|Created|2004-03-19 07:30:30Z|

Attachments:

* [persistadapt.tbz2](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/170/170377543ef02d8082756be316e863c382a1d1bc/acd4e0a4fe967c24573886cc2f3d1aa381773847.tbz2) (1461 bytes) - added by ironfroggy on 2004-03-19 07:30:30Z - 
* [persistadapt-0.2.tbz2](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/170/170377543ef02d8082756be316e863c382a1d1bc/c9ae3b47821bb5dbb0d20edb7b3e5247eedc175c.tbz2) (1470 bytes) - added by ironfroggy on 2004-03-19 07:46:07Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__556 556
type__enhancement enhancement
reporter__ironfroggy ironfroggy
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1079681430000000 1079681430000000
changetime__1475901211709912 1475901211709912
version__ 
owner__ 
cc__spiv cc__itamarst cc__ironfroggy
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ironfroggy's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ironfroggy commented|
|-|-|

```
#!html
<pre>
I had need for persistant adapters and for various reasons it was just not 
acceptable to only make it persistant at adaption time, as the current 
implementation supports. It was suggested that it would be better at 
registration (by radix, i think?), so here is a patch to allow for this. 
 
a kwarg, 'persist', may be passed to components.registerAdapter(), now and set 
to True. This will cause the adapter to be persistant by default. The attached 
file is a tar with the diff and a simple test. If anything is wrong, let me 
know about it and I'll adjust accordingly.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="ironfroggy's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ironfroggy commented|
|-|-|

```
#!html
<pre>
ah crap, my test didnt actually test the right thing! here's the package 
again, and right this time.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
Is this superseded by new interfaces?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
The whole persist concept is now deprecated. If you wish to do this implement
the logic in your adapter's logic.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc commented|
|-|-|

In changeset 49365f5c51d0e2dd6a808bfc7dee96946bba81a0
```
#!CommitTicketReference repository="" revision="49365f5c51d0e2dd6a808bfc7dee96946bba81a0"
Merge pull request #556 from twisted/8845-scop-spelling

Author: scop
Reviewer: rodrigc
Fixes: #8845

Spelling fixes
```
